### ISSUE TITLE
DYNAMIC_ARCH is broken with aarch64

### ISSUE THREAD

**Issue by paleozogt** [xianyi/OpenBLAS#2739]
With OpenBLAS v0.3.10, cross-compiling for `aarch64` is broken when `DYNAMIC_ARCH` is enabled:

```
export CC=aarch64-linux-gnu-gcc
export CXX=aarch64-linux-gnu-g++
cmake .. \
    -DCMAKE_SYSTEM_NAME=Linux \
    -DCMAKE_CROSSCOMPILING=TRUE \
    -DCMAKE_SYSTEM_PROCESSOR=aarch64 \
    -DNOFORTRAN=TRUE \
    -DDYNAMIC_ARCH=TRUE
...

make
...

[ 57%] Building ASM object kernel/CMakeFiles/kernel_CORTEXA53.dir/CMakeFiles/sgemm_otcopy_CORTEXA53.S.o
/home/paleozogt/Development/OpenBLAS/.build/kernel/CMakeFiles/sgemm_otcopy_CORTEXA53.S:8:10: fatal error: /home/paleozogt/Development/OpenBLAS/kernel/arm64/sgemm_tcopy_4.S: No such file or directory
 #include "/home/paleozogt/Development/OpenBLAS/kernel/arm64/sgemm_tcopy_4.S"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
kernel/CMakeFiles/kernel_CORTEXA53.dir/build.make:1107: recipe for target 'kernel/CMakeFiles/kernel_CORTEXA53.dir/CMakeFiles/sgemm_otcopy_CORTEXA53.S.o' failed
```

With `DYNAMIC_ARCH` disabled it builds fine.

**Comment by martin-frbg** [comment]
Ah, thanks. That got broken in CMAKE-based cross-compilation (only) when the SGEMM kernel for A53 was changed to 8x8. Guess our CI could do with a CMAKE-DYNAMIC-ARCH-ARM64. I*ll prepare a fix for cmake/prebuild.cmake in a minute.

**Comment by martin-frbg** [comment]
Unfortunately ThunderX seems to be broken in the cmake build as well (wants to link an sgemm_incopy kernel that it does not need). Probably unrelated cause but would break the DYNAMIC_ARCH build too, so I have not merged the A53 fix yet.

**Comment by martin-frbg** [comment]
Merged #2741 as the ThunderX cmake problem seems to have been introduced post 0.3.10 (and affects dedicated builds for that target as well). Bisecting that now.

**Comment by martin-frbg** [comment]
ThunderX problem is actually irreproducible, must have been a broken checkout or other local problem so retroactively marking #2741 as the fix now
